### PR TITLE
ARM/Linux: Fix detect ucol_setMaxVariable in cmake

### PR DESCRIPTION
**Pull request by seanshpark**
Fix ucol_setVariableTop related compile error for some gcc compilers

Related Issue: #5324

### CODE REVIEW

File: src/corefx/System.Globalization.Native/configure.cmake [hunk 1]
```diff
@@ -10,6 +10,9 @@ CHECK_CXX_SOURCE_COMPILES("
 
 if(NOT CLR_CMAKE_PLATFORM_DARWIN)
     set(CMAKE_REQUIRED_LIBRARIES ${ICUUC} ${ICUI18N})
+    if(NOT CLR_CMAKE_PLATFORM_FREEBSD)
+        set(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} icudata m stdc++ dl)
+    endif()
 else()
```
Comment (jasonwilliams200OK) [inline src/corefx/System.Globalization.Native/configure.cmake]: I was thinking:

```diff
      set(CMAKE_REQUIRED_LIBRARIES ${ICUUC} ${ICUI18N})
+    if(CLR_CMAKE_PLATFORM_ARCH_ARM)
+        list(APPEND CMAKE_REQUIRED_LIBRARIES icudata m stdc++ dl)
+    endif()
```

**Comment by seanshpark** [comment]
@janvorli , @mikem8361 , PTAL

File: src/corefx/System.Globalization.Native/configure.cmake [hunk 2]
```diff
@@ -10,6 +10,9 @@ CHECK_CXX_SOURCE_COMPILES("
 
 if(NOT CLR_CMAKE_PLATFORM_DARWIN)
     set(CMAKE_REQUIRED_LIBRARIES ${ICUUC} ${ICUI18N})
+    if(CLR_CMAKE_PLATFORM_UNIX_ARM)
+        set(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} icudata m stdc++ dl)
```
Comment (janvorli) [inline src/corefx/System.Globalization.Native/configure.cmake]: Is it really necessary to have the m, stdc++ and dl here?

Comment (seanshpark) [inline src/corefx/System.Globalization.Native/configure.cmake]: Not all but with gcc-5.3(gcc linker is used) it fails to link as described in https://github.com/dotnet/coreclr/issues/5324#issuecomment-222857590 . I hope there can be better way than check for ARM and add libs.

Comment (jasonwilliams200OK) [inline src/corefx/System.Globalization.Native/configure.cmake]: @seanshpark, can you reproduce this issue on other arm devices with gcc-5.3? Or could it be the local misconfiguration (some bad exports etc.) requiring this? Would be nice if you could reconfirm.

**Comment by seanshpark** [comment]
@jasonwilliams200OK , You are right. It seems my personal configuration problem. I've prepared a fresh ubuntu 15.10 + clang-3.8 (built from source) + ubuntu default arm-linux-gnueabi-gcc v5.2.1 (but not 5.3) and it is working. I'll close this issue and go back to issue #5324 .

**Pull request closed without merging**